[Bedrock] Added Amazon Bedrock examples

AntonP84 opened the pull request:
Two examples are added for Amazon Bedrock into `integrations/amazon_bedrock/examples` folder:

1. `chatgenerator_example.py` for `AmazonBedrockChatGenerator`
2. `embedders_generator_with_rag_example.py` for `AmazonBedrockGenerator`, `AmazonBedrockDocumentEmbedder` and `AmazonBedrockTextEmbedder` within a RAG pipeline

Resolves #475

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/deepset-ai/haystack-core-integrations?pullRequest=635) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/deepset-ai/haystack-core-integrations?pullRequest=635) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/deepset-ai/haystack-core-integrations?pullRequest=635) it.</sub>

vblagoje commented:
Thanks for this contribution @AntonP84 I will take it from here.

vblagoje commented:
Looks good @AntonP84 The only feedback I have is that perhaps it would be nice to use system chat message. We have internal handling that properly setups system parameter in Claude. That way examples are more portable across generators 🙏

AntonP84 commented:
Thank you @vblagoje  for the feedback. I reviewed `AnthropicClaudeChatAdapter` - the implementation turned out to be better than I expected. I moved the system prompt to messages and added a comment about it.

vblagoje reviewed (approved):
Let's add these, thanks for your contribution @AntonP84
